src/WebJobs.Script/Utility.cs

```diff
@@ -439,17 +439,18 @@ internal static bool TryReadFunctionConfig(string scriptDir, out string json, IF
             return true;
         }
 
-        internal static bool IsSingleLanguage(IEnumerable<FunctionMetadata> functions, string language)
+        internal static bool IsSingleLanguage(IEnumerable<FunctionMetadata> functions, string currentRuntimeLanguage)
         {
-            if (string.IsNullOrEmpty(language))
+            var functionsListWithoutProxies = functions?.Where(f => f.IsProxy == false);
+            if (functionsListWithoutProxies == null || functionsListWithoutProxies?.Count() == 0)
```

Comment:
We shouldn't need to account for the `null` `functions` argument since, by tracking back the calls, I see that it is currently initialized as `new Collection<FunctionMetadata>`. I suppose we could be defensive within the code and do `null` checking, but since this this `internal` only, I've removed the `null` check. 

I've used `ToArray()` to convert the `functionsListWithoutProxies` to an array at the start of the function call. Do we expect this to make a significant perf difference in the subsequent three usages of `functionsListWithoutProxies` than, say, using a `List` would? Trying to understand the rationale behind using different data structures here.